drivers/usb/device/usb_dc_sam0.c

```diff
@@ -262,22 +272,25 @@ int usb_dc_attach(void)
 int usb_dc_detach(void)
 {
 	UsbDevice *regs = &REGS->DEVICE;
+	struct usb_sam0_data *data = usb_sam0_get_data();
 
 	regs->CTRLB.bit.DETACH = 1;
 	usb_sam0_wait_syncbusy();
-
+	data->cb(USB_DC_DISCONNECTED, NULL);
 	return 0;
 }
 
 /* Remove the interrupt and reset the device */
 int usb_dc_reset(void)
 {
 	UsbDevice *regs = &REGS->DEVICE;
+	struct usb_sam0_data *data = usb_sam0_get_data();
 
 	irq_disable(DT_INST_IRQN(0));
 
 	regs->CTRLA.bit.SWRST = 1;
 	usb_sam0_wait_syncbusy();
+	data->cb(USB_DC_RESET, NULL);
```

Comment:
Is this necessary? Other platform drivers do not seem to do this.

Also EORST interrupt should be triggered when reset is done anyway.